tests/e2e/testutils.go

```diff
@@ -186,8 +188,57 @@ func ParsePods(kubeconfig string, debug bool) ([]Pod, error) {
 		}
 		pods = append(pods, pod)
 	}
-	if debug {
+	if print {
 		fmt.Println(podList)
 	}
 	return pods, nil
 }
+
+//Runs command passed from within the node
+func RunCmdOnNode(cmd string, nodename string) (string, error) {
+	runcmd := "vagrant ssh -c " + cmd + " " + nodename
+	out, err := RunCommand(runcmd)
+	if err != nil {
+		return "", err
+	}
+	return out, nil
+
+}
+
+// RunCommand Runs command on the cluster accessing the cluster through kubeconfig file
+func RunCommand(cmd string) (string, error) {
+	c := exec.Command("bash", "-c", cmd)
+	var out bytes.Buffer
+	c.Stdout = &out
+	if err := c.Run(); err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
+func UpgradeCluster(serverNodenames []string, agentNodenames []string) error {
+	for _, element := range serverNodenames {
+		nodeName := strings.TrimSpace(element)
+		if nodeName == "" {
+			continue
+		}
```

Comment:
We should need to trim or check for empty server node names.
```suggestion
func UpgradeCluster(serverNodenames []string, agentNodenames []string) error {
	for _, nodeName := range serverNodenames {
```